spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mqtt.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.util.Assert;
+
+/**
+ * @param <T> MQTT client type
+ *
+ * @author Artem Vozhdayenko
+ *
+ * @since 6.0
+ */
+public abstract class AbstractMqttClientManager<T> implements ClientManager<T>, ApplicationEventPublisherAware {
+
+	protected final Log logger = LogFactory.getLog(this.getClass()); // NOSONAR
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private boolean manualAcks;
+
+	private String url;
+
+	private final String clientId;
+
+	volatile T client;
```

Comment:
Sorry, not sure I understand, which Sonar rule do you mean?
(I don't have access to spring sonar rules and use default ones - the only Sonar issue I have is that volatile is not enough for the reference and better use `AtomicRef`. However it should not be relevant in our case, because all mutations are done in synchronized sections.)

Regarding `proptected`, I made it package-private just to limit the visibility by this package, not to expose more than needed. Do you have some practice or standard making all things protected instead of default visibility when needed in inheritor classes?

(I can change it, no problem, just my interest).

Reply:
Ah, I think I see your point - to have private field and protected synchronized setter.

Reply:
Having it `private` will satisfy a mutation vulnerability.
Mostly we don't use `package protected`, but rather `protected`: imaging that you'd like to have your own extension of this class, so such a `package protected` is not going to be visible over there in your package.